theano/tensor/tests/test_opt.py

```diff
@@ -5878,19 +5878,28 @@ def test_local_useless_split():
 
 
 def test_local_flatten_lift():
+    """
+    .. note:: The Flatten(Op) is deprecated, and this method
+        should be removed with Flatten.
+    """
     for i in xrange(1, 4):
-        op = tensor.Flatten(i)
         x = tensor.tensor4()
-        out = op(T.exp(x))
+        out = tensor.flatten(T.exp(x), i)
         assert out.ndim == i
         mode = compile.mode.get_default_mode()
-        mode = mode.including('local_flatten_lift')
+        mode = mode.including('local_reshape_lift')
         f = theano.function([x], out, mode=mode)
-        f(numpy.random.rand(5, 4, 3, 2).astype(config.floatX))
+        x_np = numpy.random.rand(5, 4, 3, 2).astype(config.floatX)
+        out_np = f(x_np)
         topo = f.maker.fgraph.toposort()
-        assert len(topo) == 2
-        assert isinstance(topo[0].op, tensor.Flatten)
-        assert isinstance(topo[1].op, tensor.Elemwise)
+        shape_out_np = tuple(x_np.shape[:i-1])+(numpy.prod(x_np.shape[i-1:]),)
+        assert shape_out_np == out_np.shape
+
+        reshape_nodes = filter(
+            lambda apply_node: isinstance(apply_node.op, tensor.Reshape), topo)
```

Comment:
Si, this does not work in Python 3, because `filter` does not return a list but a generator object, which has no length.
One option would be to explicitly call `list(...)` on the result, which would also work on Python 2.
Another option, which I find more "pythonic" than filter, would be to use the list comprehension syntax:
```
reshape_nodes = [n for n in topo if isinstance(n.op, tensor.Reshape)]
```